tests/integration/test_optimization_core.py

```
@@ -957,6 +958,63 @@ def test_enforce_under_limit_does_not_increment_violation_count(self) -> None:
+        ):
+            rss = MemoryLimiter._get_rss()
+
+        assert rss == 4096
+
+    def test_get_rss_returns_zero_when_proc_and_resource_unavailable(self) -> None:
+        from file_organizer.optimization.memory_limiter import MemoryLimiter
+
+        import_error = ImportError("no resource")
+        original_import = __import__
+
+        def _import_side_effect(name: str, *args: Any, **kwargs: Any) -> Any:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)